core/src/main/java/org/apache/druid/java/util/http/client/pool/ResourcePool.java

```diff
@@ -154,30 +167,19 @@ public void close()
     }
   }
 
-  private static class ImmediateCreationResourceHolder<K, V> implements Closeable
+  private static class EagerCreationResourceHolder<K, V> extends ResourceHolderPerKey<K, V>
   {
-    private final int maxSize;
-    private final K key;
-    private final ResourceFactory<K, V> factory;
-    private final ArrayDeque<ResourceHolder<V>> resourceHolderList;
-    private int deficit = 0;
-    private boolean closed = false;
-    private final long unusedResourceTimeoutMillis;
-
-    private ImmediateCreationResourceHolder(
+    private EagerCreationResourceHolder(
         int maxSize,
         long unusedResourceTimeoutMillis,
         K key,
         ResourceFactory<K, V> factory
     )
     {
-      this.maxSize = maxSize;
-      this.key = key;
-      this.factory = factory;
-      this.unusedResourceTimeoutMillis = unusedResourceTimeoutMillis;
-      this.resourceHolderList = new ArrayDeque<>();
-
-      for (int i = 0; i < maxSize; ++i) {
+      super(maxSize, unusedResourceTimeoutMillis, key, factory);
+      eagerInstantiation = true;
```

Comment:
It seems that `eagerInstantiation` is a field of the super class `ResourceHolderPerKey`. If this is the case, it must be taken as a constructor arg for that super class.

Reply:
I'm not sure if the way eagerInstantiation is being used within ResourceHolderPerKey is considered a good practice.
The reason it's not being used as a field in the constructor is to avoid potential misuse (such as eagerInstantiation = false) in the eager super class

Reply:
Perhaps a method `isEager()` will be better in this particular case?